src/R/Components/Test/InteractiveWorkflow/RInteractiveWorkflowCommandTest.cs

```diff
@@ -106,6 +106,7 @@ public class RInteractiveWorkflowCommandTest : IDisposable {
             command.Should().BeInvisibleAndDisabled();
 
             using (await UIThreadHelper.Instance.Invoke(() => _workflow.GetOrCreateVisualComponentAsync())) {
+                using (await _workflow.RSession.BeginInteractionAsync()) { }
```

Comment:
These sorts of issues were part of my motivation suggesting that `Connected` (as in, the event that's exposed outside of `RSession`) should perhaps only be fired on the first prompt, and everything before that prompt should be treated as internal initialization. What you suggest would be one way to do just that, if I understand correctly :)